assessments/projects/rook/self-assessment.md

```diff
@@ -0,0 +1,374 @@
+if a vulnerability in either one would compromise the other, then the distinction
+between the database and front-end is not relevant.
+
+The means by which actors are isolated should also be described, as this is often
+what prevents an attacker from moving laterally after a compromise.
+
+![Rook Components on Kubernetes](Rook%20High-Level%20Architecture.png)
+
+- `Rook Operator`
+	- `Ceph CSI Driver`
+	- `Ceph Daemons`
+
+#### `Rook Operator`
+The Rook operator is a simple container that has all that is needed to bootstrap and monitor the storage cluster. The operator will start and monitor Ceph monitor pods, the Ceph OSD daemons to provide RADOS storage, as well as start and manage other Ceph daemons. The operator manages CRDs for pools, object stores (S3/Swift), and filesystems by initializing the pods and other resources necessary to run the services.
+
+The operator will monitor the storage daemons to ensure the cluster is healthy. Ceph mons will be started or failed over when necessary, and other adjustments are made as the cluster grows or shrinks. The operator will also watch for desired state changes specified in the Ceph custom resources (CRs) and apply the changes.
+
+Rook automatically configures the Ceph-CSI driver to mount the storage to your pods. The rook/ceph image includes all necessary tools to manage the cluster. Rook is not in the Ceph data path. Many of the Ceph concepts like placement groups and crush maps are hidden so you don't have to worry about them. Instead, Rook creates a simplified user experience for admins that is in terms of physical resources, pools, volumes, filesystems, and buckets. Advanced configuration can be applied when needed with the Ceph tools.
+
+#### `Ceph CSI Driver`
+The Ceph-CSI driver provides the provisioning and mounting of volumes
+
+#### `Ceph Daemons`
+The Ceph daemons run the core storage architecture. See the [Glossary](https://github.com/rook/rook/blob/master/Documentation/Getting-Started/glossary.md#ceph) to learn more about each daemon.
+
+### Actions
+These are the steps that a project performs in order to provide some service
+or functionality.  These steps are performed by different actors in the system.
+Note, that an action need not be overly descriptive at the function call level.  
+It is sufficient to focus on the security checks performed, use of sensitive 
+data, and interactions between actors to perform an action.  
+
+For example, the access server receives the client request, checks the format, 
+validates that the request corresponds to a file the client is authorized to 
+access, and then returns a token to the client.  The client then transmits that 
+token to the file server, which, after confirming its validity, returns the file.
+
+Rook can be used to automatically configure the Ceph CSI drivers to mount the storage to an application's pods. See image at top of [Actors](#actors) for reference
+
+##### Actors
```

Comment:
Kindly elaborate on the roles of each of these actors. A good example is [Actors - Pixie Self assessment](https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#actors)

Reply:
Generic content removed and actor section significantly expanded in this commit: [fe0d055](https://github.com/dhauss/tag-security/commit/fe0d055e96a6fa26851fc13a7d9d5ed33aa98ed4)